src/ui/map.js

```diff
@@ -1824,6 +1824,22 @@ class Map extends Camera {
         return this.style.querySourceFeatures(sourceId, parameters);
     }
 
+    /**
+     * Determines if the given point is located on a visible map surface.
+     *
+     * @param {PointLike} point - The point to be checked, specified as an array of two numbers representing the x and y coordinates, or as a {@link https://docs.mapbox.com/mapbox-gl-js/api/geography/#point|Point} object.
+     * @returns {boolean} Returns `true` if the point is on the visible map surface, otherwise returns `false`.
+     * @example
+     * const pointOnSurface = map.isPointOnSurface([100, 200]);
+     */
+    isPointOnSurface(point: PointLike): boolean {
+        if (this.transform.projection.name !== 'globe' || this.transform.projection.name !== 'mercator') {
+            warnOnce(`${this.transform.projection.name} projection do not support isPointOnSurface API, this API may behave unexpectedly.`);
```

Comment:
Nit: verbose repeated chunks of code are better as a variable.

```suggestion
        const {name} = this.transform.projection;
        if (name !== 'globe' || name !== 'mercator') {
            warnOnce(`${name} projection does not support isPointOnSurface, this API may behave unexpectedly.`);
```